File: Restaurant/Restaurant.Web/Modules/ServerTypes/Default/MealsColumns.ts
import { ImageViewerFormatter } from "@/Default/Product/ImageViewerFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MealsRow } from "./MealsRow";

export interface MealsColumns {
    Id: Column<MealsRow>;
    MealsName: Column<MealsRow>;
    Photo: Column<MealsRow>;
    Description: Column<MealsRow>;
    Price: Column<MealsRow>;
    PriceBefor: Column<MealsRow>;
    Currency: Column<MealsRow>;
}

export class MealsColumns extends ColumnsBase<MealsRow> {
    static readonly columnsKey = 'Default.Meals';
    static readonly Fields = fieldsProxy<MealsColumns>();
}

[ImageViewerFormatter]; // referenced types

File: Restaurant/Restaurant.Web/Modules/Default/Meals/MealsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Restaurant.Default.Columns;

[ColumnsScript("Default.Meals")]
[BasedOnRow(typeof(MealsRow), CheckNames = true)]
public class MealsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int Id { get; set; }
    [EditLink]
    public string MealsName { get; set; }
    [ImageViewerFormatter]
    public string Photo { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public decimal PriceBefor { get; set; }
    public string Currency { get; set; }
}